src/ol_concourse/pipelines/open_edx/edx_platform_v2/earthly_packer_pulumi_pipeline.py

```diff
@@ -0,0 +1,370 @@
+                                Output(name=Identifier("edx_platform")),
+                                Output(name=Identifier("version")),
+                            ],
+                            run=Command(
+                                path="sh",
+                                args=[
+                                    "-xc",
+                                    f"""cp -r {release_name}-{deployment_name}-theme collected_themes/{deployment_name};
+                                        cp -rT {release_name}-{deployment_name}-edx-platform edx_platform;""",  # noqa: E501
+                                ],
+                            ),
+                        ),
+                    ),
+                    TaskStep(
+                        task=Identifier("build"),
+                        privileged=True,
+                        config=TaskConfig(
+                            platform=Platform.linux,
+                            image_resource=AnonymousResource(
+                                type="registry-image",
+                                source={"repository": "ardiea/earthly", "tag": "v0.3"},
+                            ),
+                            # Use some cleverness with path to mount resources within
+                            # the earthly git resource so code is where the Earthfile
+                            # expects
+                            inputs=[
+                                Input(name=earthly_git_resource.name),
+                                Input(
+                                    name=Identifier("collected_themes"),
+                                    path=f"{earthly_git_resource.name}/dockerfiles/openedx-edxapp/collected_themes",
+                                ),
+                                Input(
+                                    name=Identifier("edx_platform"),
+                                    path=f"{earthly_git_resource.name}/dockerfiles/openedx-edxapp/edx_platform",
+                                ),
+                            ],
+                            outputs=[Output(name=Identifier("artifacts"))],
+                            run=Command(
+                                path="bash",
+                                args=[
+                                    "-xc",
+                                    f"""source /docker-lib.sh;
+                                    start_docker;
+                                    echo "{release_name}-{deployment_name}-$(cat ol-infrastructure-docker/dockerfiles/openedx-edxapp/edx_platform/.git/short_ref)" > artifacts/tag.txt;
+                                    cd {earthly_git_resource.name}/dockerfiles/openedx-edxapp;
+                                    RELEASE_NAME={release_name};
+                                    DEPLOYMENT_NAME={deployment_name};
+                                    earthly +docker-image --DEPLOYMENT_NAME=$DEPLOYMENT_NAME --RELEASE_NAME=$RELEASE_NAME;
```

Comment:
The digest that we retrieve here does NOT end up being the digest that is reported by dockerhub for whatever reason that I can't be asked to understand. So for purposes of building the AMI later in the pipeline, we get the digest from proper `registry_image` Put/Get. I suspect the Get part of that is happening one more times than really necessary but I haven't found the correct optimization for that yet.